src/goto-symex/symex_clean_expr.cpp

```diff
@@ -213,8 +212,8 @@ void goto_symext::lift_lets(statet &state, exprt &rhs)
   }
 }
 
-void goto_symext::clean_expr(
-  exprt &expr,
+exprt goto_symext::clean_expr(
```

Comment:
Could you please introduce a macro wrapping `__attribute__((warn_unused_result))` - I thought that @smowton had already done this, but now I can't find it. Otherwise changes of this kind result in bugs that are terrible to debug.

Reply:
It was on an abandoned branch of the let-expr work, but it's just a port of test-gen's `NODISCARD` macro so should be easily recreated